services/smashGGService.js

```diff
@@ -0,0 +1,117 @@
+var request = require('request');
+var pmx = require('pmx');
+
+// example URL: https://smash.gg/tournament/salt-in-the-city-3/brackets/11558/10844/38067
+// 11558 is the event, 10844 is the phase, and 38067 is the phase group
+// returns
+function parseUrl(url) {
+    var offset = 0;
+    url = url.trim();
+    if (url.endsWith('/')) {
+        url = url.substr(0, url.length - 1)
+    }
+    if (url.endsWith('/bracket')) {
+        offset = 1;
+    }
+    splitUrl = url.split('/');
+    return {
+        event: splitUrl[splitUrl.length - offset - 3],
+        phase: splitUrl[splitUrl.length - offset - 2],
+        phaseGroup: splitUrl[splitUrl.length - offset - 1],
+    };
+}
+
+function getPhaseGroupUrl(phaseGroupId) {
+    var prefix = "https://smash.gg/api/-/resource/gg_api./phase_group/";
+    var params = ";expand=%5B%22sets%22%2C%22seeds%22%2C%22entrants%22%5D;force=true;reset=false";
```

Comment:
This looks a little brittle, but given we're not reading from an API, it's probably the best we can do. I'll see if Mike will help us test.